### PR TITLE
fix(nx-plugin): allow some customization when running nx-plugins

### PR DESCRIPTION
**Pull request by AgentEnder**
fix(nx-plugin): allow customizing some settings from base tsconfig

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by Coly010 — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

**Review by barbados-clemens — APPROVED** [review]

File: packages/nx/src/utils/nx-plugin.ts [hunk 1]
```diff
@@ -197,17 +199,27 @@ export function registerPluginTSTranspiler() {
   if (!tsNodeAndPathsRegistered) {
     // nx-ignore-next-line
     const ts: typeof import('typescript') = require('typescript');
```
Comment (meeroslav) [inline packages/nx/src/utils/nx-plugin.ts:201]: Why are we importing Typescript twice?

File: packages/nx/src/utils/nx-plugin.ts [hunk 2]
```diff
@@ -197,17 +199,27 @@ export function registerPluginTSTranspiler() {
...
+      : {};
 
     registerTsConfigPaths(join(workspaceRoot, 'tsconfig.base.json'));
     registerTranspiler({
       lib: ['es2021'],
       module: ts.ModuleKind.CommonJS,
       target: ts.ScriptTarget.ES2021,
       inlineSourceMap: true,
-      esModuleInterop: true,
+      esModuleInterop: tsConfig.options.esModuleInterop ?? false,
```
Comment (meeroslav) [inline packages/nx/src/utils/nx-plugin.ts:217]: This would break if there is no `tsConfigName` and `tsConfig` gets value from line 209 `{}`

Comment (meeroslav) [reply packages/nx/src/utils/nx-plugin.ts:217]: The same goes for the following `tsConfig.options.*` checks

**Review by meeroslav — DISMISSED** [review]
Left some minor comments for the nx-plugin.ts

File: packages/nx/src/utils/nx-plugin.ts (hunk 2 above)
Comment (AgentEnder) [reply packages/nx/src/utils/nx-plugin.ts:217]: Yes, it should take advantage of nullish coalescence. Thanks, will fix.

File: packages/nx/src/utils/nx-plugin.ts (hunk 1 above)
Comment (AgentEnder) [reply packages/nx/src/utils/nx-plugin.ts:201]: The line 28 should **_not_** be there. I'll get it removed.

File: packages/nx/src/utils/nx-plugin.ts [hunk 3]
```diff
@@ -197,17 +199,27 @@ export function registerPluginTSTranspiler() {
...
+      ? 'tsconfig.base.json'
+      : existsSync('tsconfig.json')
+      ? 'tsconfig.json'
+      : null;
+    const tsConfig: Partial<ts.ParsedCommandLine> = tsConfigName
+      ? readTsConfig(tsConfigName)
+      : {};
 
     registerTsConfigPaths(join(workspaceRoot, 'tsconfig.base.json'));
     registerTranspiler({
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/nx-plugin.ts:212]:
```suggestion
    registerTranspiler({
      esModuleInterop: false,
      experimentalDecorators: true,
      ...tsConfig.options,
      inlineSourceMap: true,
```

**Review by FrozenPandaz — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**